src/lib.rs

```diff
@@ -158,21 +158,177 @@ mod core {
 use core::fmt;
 
 #[cfg(not(feature = "std"))]
+#[macro_use]
 extern crate alloc;
+
 #[cfg(not(feature = "std"))]
 use alloc::Vec;
+#[cfg(not(feature = "std"))]
+use alloc::string::String;
+#[cfg(not(feature = "std"))]
+use alloc::borrow::ToOwned;
+#[macro_use]
+extern crate serde_derive;
+
+extern crate serde;
+
+use serde::ser::{SerializeStruct, SerializeSeq, Serialize, Serializer};
+
+/// Types allowed to be value in the context vector
+#[derive(Debug, Clone)]
+pub enum ThrowContextValues {
+    ///Boolean
+    Bool(bool),
+    ///Int8
+    Int8(i8),
+    ///Uint8
+    Uint8(u8),
+    ///Int16
+    Int16(i16),
+    ///Uint16
+    Uint16(u16),
+    ///Int32
+    Int32(i32),
+    ///Uint32
+    Uint32(u32),
+    ///Int64
+    Int64(i64),
+    ///Uint64
+    Uint64(u64),
+    ///Float32
+    Float32(f32),
+    ///Float64
+    Float64(f64),
+    ///String
+    String(String),
+}
+
+impl Serialize for ThrowContextValues {
+    fn serialize<S>(&self, serializer: S) -> core::result::Result<S::Ok, S::Error>
+        where S: Serializer
+    {
+        match self {
+            &ThrowContextValues::Bool(ref x) => { return Ok(serializer.serialize_bool(*x)?); }
+            &ThrowContextValues::Int8(ref x) => { return Ok(serializer.serialize_i8(*x)?); }
+            &ThrowContextValues::Uint8(ref x) => { return Ok(serializer.serialize_u8(*x)?); }
+            &ThrowContextValues::Int16(ref x) => { return Ok(serializer.serialize_i16(*x)?); }
+            &ThrowContextValues::Uint16(ref x) => { return Ok(serializer.serialize_u16(*x)?); }
+            &ThrowContextValues::Int32(ref x) => { return Ok(serializer.serialize_i32(*x)?); }
+            &ThrowContextValues::Uint32(ref x) => { return Ok(serializer.serialize_u32(*x)?); }
+            &ThrowContextValues::Int64(ref x) => { return Ok(serializer.serialize_i64(*x)?); }
+            &ThrowContextValues::Uint64(ref x) => { return Ok(serializer.serialize_u64(*x)?); }
+            &ThrowContextValues::Float32(ref x) => { return Ok(serializer.serialize_f32(*x)?); }
+            &ThrowContextValues::Float64(ref x) => { return Ok(serializer.serialize_f64(*x)?); }
+            &ThrowContextValues::String(ref x) => { return Ok(serializer.serialize_str(x)?); }
+        };
+    }
+}
+
+
+impl fmt::Display for ThrowContextValues {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            &ThrowContextValues::Bool(ref x) => { write!(f, "{}", x) }
```

Comment:
Not a big deal, but it's nicer to do `match *self {` here than add `&` on every branch. (both here and in the `Serialize`) impl.

For the integer types, `ref` should also be unnecessary. It makes sense in the String branch, but all the integer types are directly copyable.

Reply:
done